### DISCUSSION TITLE
anyone interested in a devcontainer ?

### DISCUSSION THREAD

**Discussion by jmls** [openbao/openbao#579] [General]
I've put together a small patch to add devcontainer support with docker into bao development. This allows for you to start a go environment without having to install go. 

Having docker support enables you to fire up any other container you need within the openbao environment

If there's interest from the core team to include this in the repository,  I'll submit a PR

**Comment by jmls** [comment]
I should have said for vsCode users  - I don't know if other IDE support devcontainers or not

**Reply by cipherboy** [reply]
\o hey @jmls -- I think we already have a dev container here: https://github.com/openbao/openbao/blob/main/scripts/docker/Dockerfile

This is usually accessible via `make docker-dev`: https://github.com/openbao/openbao/blob/main/Makefile#L52-L56

But if there's something this container doesn't do, happy to take additions or modifications to it!

**Comment by jmls** [comment]
I've had a look there - but I think that's for building a docker image. Unless I am wrong :) 

The devcontainer allows vsCode to start a development workspace with an image that already has go installed, so you don't need to install anything. it does not build openbao, but provides a go env to build it

**Reply by cipherboy** [reply]
Ahh, TIL: https://containers.dev/ 

Yes, feel free to add something :-)
